python/pyarrow/tests/parquet/test_parquet_file.py

```diff
@@ -177,8 +178,12 @@ def test_parquet_file_pass_directory_instead_of_file(tempdir):
     path = tempdir / 'directory'
     os.mkdir(str(path))
 
-    with pytest.raises(IOError, match="Expected file path"):
+    msg = f"Cannot open for reading: path '{str(path)}' is a directory"
+    with pytest.raises(IOError) as exc:
         pq.ParquetFile(path)
+    if exc.errisinstance(PermissionError) and sys.platform == 'win32':
+        return  # Windows CI can get a PermissionError here.
```

Comment:
cc @pitrou do you have any idea why trying to open a directory on windows gives a PermissionError instead of a normal IO error with the informative message?

Reply:
I suspect it's just the normal behaviour on Windows... is memory-mapping activated?

Reply:
> is memory-mapping activated?

That's optional, with the default of False, so it shouldn't be used in this case